test/battle/move_effect/doodle.c

```diff
@@ -0,0 +1,26 @@
+#include "global.h"
+#include "test/battle.h"
+
+ASSUMPTIONS
+{
+    ASSUME(gBattleMoves[MOVE_DOODLE].effect == EFFECT_DOODLE);
+}
+
+DOUBLE_BATTLE_TEST("Doodle gives the target's ability to user and ally")
```

Comment:
Can you write a/some tests that show Doodle failing if any three of the Pokémon involved have a forbidden ability? `PARAMETRIZE` will be useful for this.

Also can we get a `SINGLE_BATTLE_TEST` for whatever the expected behavior of Doodle in singles is.

Reply:
Doodle in singles is exactly the same as Role Play.

Reply:
Cool! So in that case I'd copy Role Play's applicable single test(s) and change them to use Doodle.

Or alternatively I suppose you could `PARAMETRIZE` those applicable tests over both `MOVE_ROLE_PLAY` and `MOVE_DOODLE`.